prometheus/promhttp/delegator.go

```diff
@@ -61,8 +61,10 @@ func (r *responseWriterDelegator) WriteHeader(code int) {
 		r.observeWriteHeader(code)
 	}
 	r.status = code
-	r.wroteHeader = true
-	r.ResponseWriter.WriteHeader(code)
```

Comment:
As described in the comment above (lines 57 to 60), we intentionally delegate the `WriteHeader` call to not mask the actual bug. Correct code wouldn't call `WriteHeader` more than once in the first place.